erpnext/public/js/controllers/taxes_and_totals.js

```diff
@@ -390,10 +390,21 @@ erpnext.taxes_and_totals = erpnext.payments.extend({
 		this.frm.doc.grand_total = flt(tax_count
 			? this.frm.doc["taxes"][tax_count - 1].total + flt(this.frm.doc.rounding_adjustment)
 			: this.frm.doc.net_total);
-
+		var payment_status = true;
 		if(in_list(["Quotation", "Sales Order", "Delivery Note", "Sales Invoice"], this.frm.doc.doctype)) {
 			this.frm.doc.base_grand_total = (this.frm.doc.total_taxes_and_charges) ?
 				flt(this.frm.doc.grand_total * this.frm.doc.conversion_rate) : this.frm.doc.base_net_total;
+			if (this.frm.doc.is_pos && this.frm.doc.is_return) {
```

Comment:
I think this should be handled in `set_default_payment` method only (by modifying if condition). 
https://github.com/frappe/erpnext/blob/hotfix/erpnext/public/js/controllers/taxes_and_totals.js#L619

And then the method should be called from https://github.com/frappe/erpnext/blob/hotfix/erpnext/public/js/controllers/taxes_and_totals.js#L574

Reply:
The problem here is that if we add code in `set_default_payment()` and call this in `calculate_outstanding_amount()` then `calculate_outstanding_amount()` is called when user manually change Amount in Payments Child Table. Method doesn't get triggered when Item qty is being updated in Item Child Table to update Payment Amount respectively in Payments Child Table.

Where as we want Amount in Payments Child Table to be updated as soon as Item qty is being updated in Item Child Table which is working as we have added code to `calculate_totals()`.

Reply:
The problem here is that if we add code in `set_default_payment()` and call this in `calculate_outstanding_amount()` then `calculate_outstanding_amount()` is called when user manually change Amount in Payments Child Table. Method doesn't get triggered when Item qty is being updated in Item Child Table to update Payment Amount respectively in Payments Child Table.

Where as we want Amount in Payments Child Table to be updated as soon as Item qty is being updated in Item Child Table which is working as we have added code to `calculate_totals()`.

Reply:
ok